cli/src/main/java/com/devonfw/tools/ide/io/IdeProgressBarConsole.java

```diff
@@ -1,33 +1,88 @@
 package com.devonfw.tools.ide.io;
 
+import com.devonfw.tools.ide.os.SystemInfo;
+
 import me.tongfei.progressbar.ProgressBar;
+import me.tongfei.progressbar.ProgressBarBuilder;
+import me.tongfei.progressbar.ProgressBarStyle;
 
 /**
  * Implementation of {@link IdeProgressBar}.
  */
-public class IdeProgressBarConsole implements IdeProgressBar {
+public class IdeProgressBarConsole extends AbstractIdeProgressBar {
 
   private final ProgressBar progressBar;
+  private final SystemInfo systemInfo;
 
   /**
    * The constructor.
    *
-   * @param progressBar the {@link ProgressBar} to initialize.
+   * @param systemInfo the {@link SystemInfo}.
+   * @param taskName the {@link ProgressBar} to initialize.
+   * @param maxSize the maximum size of the progress bar.
+   */
+  public IdeProgressBarConsole(SystemInfo systemInfo, String taskName, long maxSize) {
+
+    super(maxSize);
+
+    this.systemInfo = systemInfo;
+    this.progressBar = createProgressBar(taskName, maxSize);
+  }
+
+  /**
+   * Creates the {@link ProgressBar} initializes task name and maximum size as well as the behaviour and style.
+   *
+   * @param taskName name of the task.
+   * @param size of the content.
+   * @return {@link ProgressBar} to use.
    */
-  public IdeProgressBarConsole(ProgressBar progressBar) {
+  protected ProgressBar createProgressBar(String taskName, long size) {
 
-    this.progressBar = progressBar;
+    ProgressBarBuilder pbb = new ProgressBarBuilder();
+    // default (COLORFUL_UNICODE_BLOCK)
+    pbb.setStyle(ProgressBarStyle.builder().refreshPrompt("\r").leftBracket("\u001b[33m│").delimitingSequence("")
+        .rightBracket("│\u001b[0m").block('█').space(' ').fractionSymbols(" ▏▎▍▌▋▊▉").rightSideFractionSymbol(' ')
+        .build());
+    // set different style for Windows systems (ASCII)
+    if (this.systemInfo.isWindows()) {
+      pbb.setStyle(ProgressBarStyle.builder().refreshPrompt("\r").leftBracket("[").delimitingSequence("")
+          .rightBracket("]").block('=').space(' ').fractionSymbols(">").rightSideFractionSymbol(' ').build());
+    }
```

Comment:
Now that we have this code in review again:
It would be better to set the default stype in an `else` block. You are first stetting the style and then on windows setting it again overriding it.
Or even better since both variants share the same code fragment:
```suggestion
    String leftBracket, rightBracket, fractionSymbols;
    char block;
    if (this.systemInfo.isWindows()) {
      // set different style for Windows systems (ASCII)
      leftBracket = "[";
      rightBracket = "]";
      fractionSymbols = ">";
      block = '=';
    } else {
      // default (COLORFUL_UNICODE_BLOCK)
      leftBracket = "\u001b[33m│";
      rightBracket = "│\u001b[0m";
      fractionSymbols = " ▏▎▍▌▋▊▉";
      block = '█';
    }
    pbb.setStyle(ProgressBarStyle.builder().refreshPrompt("\r").leftBracket(leftBracket).delimitingSequence("")
        .rightBracket(rightBracket).block(block).space(' ').fractionSymbols(fractionSymbols).rightSideFractionSymbol(' ')
        .build()); 
```